infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -34,8 +30,11 @@
      * <p>
      * a cache of recent conflict keys and their latest global-log position.
      */
-    private final Cache<ConflictTxStream, Long> conflictCache;
-
+    //As the sequencer cache is used by a single thread, it is safe to use hashmap.
+    private final HashMap<ConflictTxStream, Long> cacheHash;
+    private final PriorityQueue<ConflictTxStreamEntry> cacheEntries; //sorted according to address
```

Comment:
I suggest you remove one type. You can have ConflictKeyEntry that also encapsulates the long and can also have a comparator .